tests/stubtest_stdlib.py

```diff
@@ -6,55 +6,98 @@
 In typeshed CI, we run stubtest with each currently supported Python minor version.
 
 """
+from __future__ import annotations
 
 import subprocess
 import sys
+import tempfile
 from pathlib import Path
 
+from pkg_resources import parse_requirements
+
+from utils import make_venv, print_error
+
 
 def run_stubtest(typeshed_dir: Path) -> int:
     allowlist_dir = typeshed_dir / "tests" / "stubtest_allowlists"
     version_allowlist = f"py{sys.version_info.major}{sys.version_info.minor}.txt"
     platform_allowlist = f"{sys.platform}.txt"
     combined_allowlist = f"{sys.platform}-py{sys.version_info.major}{sys.version_info.minor}.txt"
+    with tempfile.TemporaryDirectory() as tmp:
+        venv_dir = Path(tmp)
+        try:
+            pip_exe, python_exe = make_venv(venv_dir)
+        except Exception:
+            print_error("fail")
+            raise
+
+        with Path("requirements-tests.txt").open() as requirements_txt:
+            requirements = [str(requirement) for requirement in parse_requirements(requirements_txt) if requirement.key == "mypy"]
```

Comment:
I'd prefer it if we used the [`packaging.requirements.Requirement` class](https://packaging.pypa.io/en/stable/requirements.html#requirements) for this parsing logic, since the `pkg_resources` library is deprecated. (I looked into whether we could replace our usage of `pkg_resources` in `pytype_test.py`, but I concluded that it wasn't worth it -- we'd have to add `importlib_metadata` as a dependency in order for our tests to run on <3.10.)

Reply:
I can't find how to parse a `requirements.txt` file using `packaging.requirements.Requirement` without also having to manually remove comments. Sure the `mypy` line doesn't have comments right now. But if it does, I would like if it didn't break.

Reply:
`Requirement(line.split("#")[0].strip())`? :)

Reply:
That's what I ended up going for too